File: bot/src/main/java/com/yauhenl/pibot/service/ActionService.java
package com.yauhenl.pibot.service;

import com.yauhenl.pibot.control.action.ActionType;

/**
 * Created by Stsiapan_Shablinski on 7/28/2016.
 */
public class ActionService {

    private static ActionService actionService = new ActionService();

    private ActionService() {
    }

    public static ActionService getInstance() {
        return actionService;
    }

    public void performAction(ActionType actionType) {
        actionType.getAction().perform(BotControl.getInstance());
    }
}

File: bot/src/main/kotlin/com/yauhenl/pibot/service/ActionService.kt
package com.yauhenl.pibot.service

import com.yauhenl.pibot.control.ActionType
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ActionService {

    @Autowired
    lateinit var botControl: BotControl

    fun performAction(actionType: ActionType) = actionType.action.invoke(botControl)
}
